src/mapillary_client.py

```diff
@@ -0,0 +1,71 @@
+from os import getenv, getcwd
+import logging
+from multiprocessing import Pool
+from pathlib import Path
+from mapillary import interface as mly
+from requests import get, RequestException
+
+
+class MapillaryClient:
+    def __init__(self, access_token: str, log_level: int = logging.INFO, basepath: str = getcwd()) -> None:
+        self.access_token = access_token
+        mly.set_access_token(access_token)
+        self.log = logging.getLogger(self.__class__.__name__)
+        self.log.addHandler(logging.StreamHandler())
+        self.log.setLevel(log_level)
+        self.basepath = basepath
+
+    def get_images_from_coordinates(self, latitude: float, longitude: float) -> list[Path]:
+        self.log.info("%s.%s: %s, %s",
+                      self.__class__.__name__,
+                      self.get_images_from_coordinates.__name__,
+                      latitude,
+                      longitude)
+
+        images = mly.get_image_close_to(latitude, longitude, image_type="pano").to_dict()
+        image_ids = list(map(lambda img: img["properties"]["id"], images["features"]))
+
+        self.log.info("%s.%s: got %s images close to %s, %s",
+                      self.__class__.__name__,
+                      self.get_images_from_coordinates.__name__,
+                      len(image_ids),
+                      latitude,
+                      longitude)
+        self.log.debug(image_ids)
+
+        with Pool() as pool:
+            return pool.map(self.download_image, image_ids)
+
+    def download_image(self, image_id: int) -> Path:
+        mly.set_access_token(self.access_token)
+        self.log.info("%s.%s: %s",
+                      self.__class__.__name__,
+                      self.download_image.__name__,
+                      image_id)
+        filepath = Path(self.basepath, f"{image_id}.jpg")
+        image_url = mly.image_thumbnail(str(image_id), 2048)
+        self.log.debug("%s.%s: image url for %s is %s",
+                      self.__class__.__name__,
+                      self.download_image.__name__,
+                      image_id,
+                      image_url)
+
+        try:
+            image_content = get(image_url).content
+            with open(filepath, "wb") as image_file:
+                image_file.write(image_content)
+            self.log.debug("%s.%s: image %s downloaded to %s",
+                          self.__class__.__name__,
+                          self.download_image.__name__,
+                          image_id,
+                          filepath)
+        except RequestException as e:
+            self.log.error(e)
+
+        return filepath
+
+
+if __name__ == "__main__":
+    client = MapillaryClient(getenv("MAPILLARY_API_KEY"))
```

Comment:
It would be great to have a `.env.example` file that contains ```
MAPILLARY_API_KEY=somefakeapikey
``` 
to help document that this is needed. This is a common convention that I've seen.

Reply:
would things like the default `basepath` go in that file as well?

Reply:
Yes, that is possible. Although maybe it should be renamed to `image_path` or something since the scripts will be run separately.